src/loader/cmti.ml

```diff
@@ -526,14 +526,20 @@ and read_module_type env parent label_parent mty =
         let decl =
           let open Module in
           match mexpr.mod_desc with
-          | Tmod_ident(p, _) -> Alias (Env.Path.read_module env p)
+          | Tmod_ident(p, _) ->
+            Format.eprintf "typeof mpath\n%!";
+            TypeOf (MPath (Env.Path.read_module env p))
           | _ ->
-              let mty =
-                !read_module_expr env parent label_parent mexpr
-              in
-                ModuleType mty
+              match !read_module_expr env parent label_parent mexpr with
+              | Signature [Odoc_model.Lang.Signature.Include {Include.decl = Alias p; _}] ->
+                Format.eprintf "typeof struct_include\n%!";
+                TypeOf (Struct_include p)
+              | Signature [Odoc_model.Lang.Signature.Include {Include.decl = ModuleType (Signature items); _}] ->
+                Signature items
```

Comment:
I'm not sure you need this special case. I think that:
```ocaml
module type S = sig include sig type t end end
```
and
```ocaml
module type S = module type of struct include struct type t end end
```
should be treated the same.

If we want to eliminate instances of `include sig ... end` then I think that should be done elsewhere, and you'll get the equivalent of this special case for free.

Reply:
See comment above.